src/Nncase.Core/Utilities/QuantUtility.cs

```diff
@@ -149,4 +149,320 @@ public static List<float> GetWeightsRangesByChannel(Span<float> weights, int cha
 
         return minMaxArr;
     }
+
+    public static Span<float> SquantWeights(Span<float> inputWeights, Expr inputWeightsRanges, Nncase.IR.Shape inputWeightsShape, QuantMode quantMode, int bits, bool isByChannel)
```

Comment:
Type of inputWeightsRanges should be `Tensor<float>` as this method doesn't support other types.

Reply:
Type of inputWeightsShape should be `ReadOnlySpan<int>` as this method doesn't support dynamic shapes.